### PR TITLE
chore: expose Mapbox road shield diagnostics

### PR DESCRIPTION
**Pull request by ebelo**
## Summary
- add active Mapbox Outdoors road-number-shield candidate buckets to the road-feature diagnostic
- report shield class, reflen, structure, layer, and combined signature counts in JSON/Markdown outputs
- include sample shield candidates so future road-symbol parity work is grounded in live vector-tile features

## Evidence
- Live converter audit: `debug/mapbox-outdoors-style-audit/mapbox-outdoors-v12/20260519T115412Z/audit.json`
  - removing `layout.icon-image` from the qfit-preprocessed style reduces 840 QGIS sprite warnings
  - qfit-preprocessed filters are parser-supported in the live filter probe, so shield/icon evidence is a better next diagnostic target than filter syntax
- Live all-camera road-feature diagnostic: `debug/mapbox-outdoors-road-features/all-cameras/20260519T122113Z/summary.md`
  - 62/62 tiles decoded
  - 535 qfit-preprocessed road-number-shield candidates across the seven-camera matrix
  - candidates concentrate in z8/z10, with only 7 total z14 candidates after matching qfit's z11+ `len > 2500` line-geometry shield filter
  - z17/z18 Zermatt high-detail cameras have no active shield candidates in this matrix

## Validation
- `python3 -m pytest tests/test_mapbox_outdoors_road_features.py -q --tb=short`
- `python3 -m pytest tests/ -x -q --tb=short`
- `git diff --check`

Refs #949

### CODE REVIEW

File: validation/mapbox_outdoors_road_features.py [hunk 1]
```diff
@@ -186,6 +199,49 @@ def is_oneway_arrow_candidate(feature: dict[str, object], *, tile_zoom: int | No
...
+    return None
+
+
+def is_road_number_shield_candidate(feature: dict[str, object], *, tile_zoom: int | None = None) -> bool:
+    properties = _feature_properties(feature)
+    reflen = _finite_number(properties.get("reflen"))
+    if (
+        tile_zoom is None
+        or tile_zoom < ROAD_NUMBER_SHIELD_MIN_ZOOM
+        or properties.get("class") in ROAD_NUMBER_SHIELD_EXCLUDED_CLASSES
```
[the remark this answers is not in the document]
Comment (ebelo) [reply validation/mapbox_outdoors_road_features.py:218]: I checked this against the generated qfit-preprocessed style before changing code. `_road_number_shield_layer_variants()` keeps `base_filter = _road_shield_filter_with_string_reflen_range(layer.get("filter"))`, and the processed `road-number-shield-*` layers still retain the original class guard:

`["match", ["get", "class"], ["pedestrian", "service"], false, true]`

The diagnostic exclusion mirrors that live style filter. Including `service` or `pedestrian` here would overcount features that the current qfit road-number-shield layers exclude, so I am leaving this as-is.

File: validation/mapbox_outdoors_road_features.py [hunk 2]
```diff
@@ -186,6 +199,49 @@ def is_oneway_arrow_candidate(feature: dict[str, object], *, tile_zoom: int | No
...
+        or reflen is None
+        or reflen > ROAD_NUMBER_SHIELD_MAX_REFLEN
+    ):
+        return False
```
[the remark this answers is not in the document]
Comment (ebelo) [reply validation/mapbox_outdoors_road_features.py:223]: Fixed in `62bafe4`: the shield predicate now rejects `reflen <= 0`, and the candidate-filter test covers the zero-reflen edge case.

File: validation/mapbox_outdoors_road_features.py [hunk 3]
```diff
@@ -56,6 +56,14 @@
...
 STEP_STRUCTURES = {"none", "ford", "bridge", "tunnel"}
 ONEWAY_ARROW_MIN_ZOOM = 16
+ROAD_NUMBER_SHIELD_MIN_ZOOM = 6
+ROAD_NUMBER_SHIELD_MAX_REFLEN = 6
+ROAD_NUMBER_SHIELD_EXCLUDED_CLASSES = {"pedestrian", "service"}
+ROAD_NUMBER_SHIELD_LENGTH_THRESHOLDS = (
+    (13, 1000.0),
+    (12, 2500.0),
+    (11, 5000.0),
+)
```
[the remark this answers is not in the document]
Comment (ebelo) [reply validation/mapbox_outdoors_road_features.py]: Fixed in `a34fab5`: the diagnostic now matches qfit's preprocessed shield filter by using the z11+ `len > 2500` threshold instead of the original dynamic Mapbox zoom stops. I also refreshed the live road-feature evidence and updated the PR body.

File: validation/mapbox_outdoors_road_features.py [hunk 4]
```diff
@@ -186,6 +195,38 @@ def is_oneway_arrow_candidate(feature: dict[str, object], *, tile_zoom: int | No
...
+        or properties.get("class") in ROAD_NUMBER_SHIELD_EXCLUDED_CLASSES
+        or reflen is None
+        or reflen <= 0
+        or reflen > ROAD_NUMBER_SHIELD_MAX_REFLEN
+    ):
+        return False
+    if tile_zoom < 11:
+        return _geometry_type(feature) == "Point"
+    length = _finite_number(properties.get("len"))
+    return length is not None and length > ROAD_NUMBER_SHIELD_LINE_LENGTH_MIN
```
[the remark this answers is not in the document]
Comment (ebelo) [reply validation/mapbox_outdoors_road_features.py:229]: Fixed in 6f1d8e9: z11+ shield candidates now require a line geometry before applying the qfit len > 2500 filter, and the candidate-filter test covers a high-zoom Point with reflen/len.
